lib/validations.rb

```diff
@@ -1,10 +1,89 @@
 class Validations
 
   def validate_ship_number(number)
-    number = number.to_i
-    valid_numbers = [1, 2, 3, 4]
-    if !valid_numbers.include?(number)
-      :error
+    valid_options = [1, 2, 3, 4]
```

Comment:
Currently these numbers are separate from the ships. So if the ships change, let's say you only have three instead of four, you have to remember to change the valid_options here as well. It would be better if you passed in the ships here so that you have a direct relationship to the actual ships and don't need to rely on remembering to update this.